Assets/ScriptableObjects/PlayerResource.cs

```diff
@@ -0,0 +1,48 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using System;
+
+[CreateAssetMenu(fileName = "New Player Resource", menuName = "PlayerResource")]
+
+public class PlayerResource : ScriptableObject
+{
+    [SerializeField] new string name;
+    [SerializeField] int maxValue;
+    [SerializeField] int minValue;
+    [SerializeField] int canvasChildNumber;
+    [SerializeField] string staticText;
+
+    private int currentValue;
+    private Canvas canvas;
+
+    private void OnEnable()
+    {
+        currentValue = maxValue;
+        canvas = GameObject.Find("Canvas").GetComponent<Canvas>();
```

Comment:
Hmm, this raises a red flag in my mind. ScriptableObjects exist outside of the Scene context and shouldn't be tied to any particular scene. Further, we probably don't want to be persisting Canvases with `DontDestroyOnLoad`. I wonder how this will work with the multi-scene system we'll be using for building out the game world.